docs/guides/expectations/contributing/how-to-contribute-a-new-expectation-to-great-expectations.md

```diff
@@ -2,4 +2,397 @@
+
+	For example, if you call your Expectation `expect_column_values_to_equal_three`, the file with its implementation should be: `contrib/experimental/great_expectations_experimental/expectations/expect_column_values_to_equal_three.py`
+
+3. Name the class
+
+	Within the file, update the name of the class that implements your Expectation. Classes that implement Expectations have CamelCase names (e.g., `ExpectColumnValuesToEqualThree`). The framework will automatically translate this class name into a method with the snake_case name of `expect_column_values_to_equal_three`. You will make this change in two places:
+
+	* Class declaration (search for `class Expect`)
+	* A call to `run_diagnostic` in the very end of the template (search for ``diagnostics_report = ``). Next section explains the role this code plays.
+	
+	For more style conventions that your code should follow consult our [Style Guide](/docs/contributing/contributing-style)
+
+#### 4. Run diagnostics on your Expectation.
+
+Expectations contain a self diagnostic tool that will help you during development. The simplest way to run it is to execute the file as a standalone script. Note: if you prefer, you can also run it within a notebook or IDE.
+
+````python
+python expect_column_values_to_equal_three.py
+````
+
+Running the script will execute the `run_diagnostics` method for your new class. Initially, it will just return:
+
+````python
+{
+  "description": {
+    "camel_name": "ExpectColumnValuesToEqualThree",
+    "snake_name": "expect_column_values_to_equal_three",
+    "short_description": "",
+    "docstring": ""
+  },
+  "library_metadata": {
+    "maturity": "experimental",
+    "package": "experimental_expectations",
+    "tags": [],
+    "contributors": []
+  },
+  "renderers": {},
+  "examples": [],
+  "metrics": [],
+  "execution_engines": {}
+  "test_report": [],
+  "diagnostics_report": []
+}
+````
+
+This output is a report on the completeness of your Expectation.
+
+You will repeat this step many times during developing your Expectation. `run_diagnostics` creates an easy and fast “dev loop” for you - make a small change in the code, run `run_diagnostics`, examine its output for failures and next steps.
+
+From this point on, we will start filling in the pieces of your Expectation. You don’t have to fill in all the pieces to submit your Expectation. For example, you may choose to provide only Pandas implementation. Another contributor may add a Spark implementation in a separate PR later. Expectation development can be done in bite-size pieces.
+
+:::note
+* If you prefer to do your development in Jupyter Notebook and copy your Expectation into the file after you are done, you will run run_diagnostics directly in the notebook (instead of executing the file):
+````console
+# TODO: replace the class name with your Expectation's class name
+diagnostics_report = ExpectColumnValuesToEqualThree().run_diagnostics()
+print(json.dumps(diagnostics_report, indent=2))
+````
```

Comment:
```suggestion
````python
# TODO: replace the class name with your Expectation's class name
diagnostics_report = ExpectColumnValuesToEqualThree().run_diagnostics()
print(json.dumps(diagnostics_report, indent=2))
````
```